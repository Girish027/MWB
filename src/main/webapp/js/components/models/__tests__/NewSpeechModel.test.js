import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import validationUtil from 'utils/ValidationUtil';
import apiUrls from 'utils/apiUrls';

const NewSpeechModelView = require('components/models/NewSpeechModel').default;

const props = {
  setValidModel: () => { },
  dispatch: () => { },
  reportSuccess: () => { },
  reportError: () => { },
  saveModelChanges: () => { },
  modelViewReadOnly: false,
};
let wrapper;

describe('<NewSpeechModel />', () => {
  describe('Creating an instance', () => {
    beforeAll(() => {
      wrapper = shallow(<NewSpeechModelView
        {...props}
      />);
    });

    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    beforeEach(() => {
      wrapper = shallow(<NewSpeechModelView
        {...props}
      />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });
    test('should match with basic props', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    const file = {
      name: 'config.csv',
    };
    beforeAll(() => {
      NewSpeechModelView.handleFileChoosen = jest.fn(() => 'handleFileChoosen');
      validationUtil.validateWorldClassFile = jest.fn(() => 'validateWorldClassFile');
      apiUrls.getDataUrl = jest.fn(() => 'getDataUrl');
      props.dispatch = jest.fn();
    });

    beforeEach(() => {
      wrapper = shallow(<NewSpeechModelView
        {...props}
      />);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('should call saveFile', () => {
      test('should upload the file successfully', () => {
        wrapper.instance().saveFile([file]);
        expect(wrapper.handleFileChoosen).toHaveBeenCalled;
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });

    describe('should call onclickDownload', () => {
      test('should call getDataURL', () => {
        wrapper.instance().onClickDownload();
        expect(apiUrls.getDataUrl).toHaveBeenCalled;
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });

    describe('should call rejectFile', () => {
      test('should call onChange', () => {
        wrapper.instance().rejectFile();
        expect(wrapper.props.onChange).toHaveBeenCalled;
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });

    describe('should call saveModelChanges', () => {
      test('should call saveModelChanges', () => {
        wrapper.instance().saveModelChanges();
        expect(wrapper.props.saveModelChanges).toHaveBeenCalled;
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });

    describe('should call isNameValid', () => {
      test('should validate the name successfully', () => {
        expect(wrapper.instance().isNameValid('abcd')).toBe(true);
      });
      test('should validate the name as invalid', () => {
        expect(wrapper.instance().isNameValid('')).toBe(false);
      });
    });

    describe('should call isDatasetValid', () => {
      test('should upload the config successfully', () => {
        wrapper.instance().isDatasetValid(['1234']);
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });

    describe('should call isValidated', () => {
      test('should set the model as valid', () => {
        wrapper.instance().isValidated();
        expect(wrapper.props.setValidModel).toHaveBeenCalled;
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });

    describe('should call getDatasetsLabel', () => {
      test('should display the dataset message correctly', () => {
        wrapper.instance().getDatasetsLabel();
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
    });
  });
});
