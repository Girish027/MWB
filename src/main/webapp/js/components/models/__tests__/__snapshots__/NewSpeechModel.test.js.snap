// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewSpeechModel /> Functionality: should call getDatasetsLabel should display the dataset message correctly 1`] = `
<div
  id="NewModel"
>
  <ul
    className="form-fields"
  >
    <li>
      <label>
        Model
      </label>
      <div
        style={
          Object {
            "color": "#313f54",
          }
        }
      >
         
      </div>
    </li>
    <li>
      <label>
        Version Description
      </label>
      <withTheme(Textarea)
        defaultValue=""
        disabled={false}
        name="modelDescription"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a description"
        styleOverride={
          Object {
            "width": "100%",
          }
        }
        type="text"
      />
    </li>
    <li>
      <MultipleCheckboxFilter
        allowEmpty={true}
        className="DatasetsFilter"
        disabled={false}
        label={
          <div
            style={
              Object {
                "color": "#727272",
              }
            }
          >
            Datasets*
          </div>
        }
        noOfColumns={3}
        onChange={[Function]}
        options={Array []}
        showAsGrid={false}
        showDropbox={false}
        type={null}
        value={Array []}
      />
    </li>
  </ul>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginBottom": "30px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "30px",
        "width": "400px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <Memo(withTheme(Download))
            fill="#004c97"
            height="13px"
            width="10px"
          />
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={false}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    />
  </div>
</div>
`;

exports[`<NewSpeechModel /> Functionality: should call isDatasetValid should upload the config successfully 1`] = `
<div
  id="NewModel"
>
  <ul
    className="form-fields"
  >
    <li>
      <label>
        Model
      </label>
      <div
        style={
          Object {
            "color": "#313f54",
          }
        }
      >
         
      </div>
    </li>
    <li>
      <label>
        Version Description
      </label>
      <withTheme(Textarea)
        defaultValue=""
        disabled={false}
        name="modelDescription"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a description"
        styleOverride={
          Object {
            "width": "100%",
          }
        }
        type="text"
      />
    </li>
    <li>
      <MultipleCheckboxFilter
        allowEmpty={true}
        className="DatasetsFilter"
        disabled={false}
        label={
          <div
            style={
              Object {
                "color": "#727272",
              }
            }
          >
            Datasets*
          </div>
        }
        noOfColumns={3}
        onChange={[Function]}
        options={Array []}
        showAsGrid={false}
        showDropbox={false}
        type={null}
        value={Array []}
      />
    </li>
  </ul>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginBottom": "30px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "30px",
        "width": "400px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <Memo(withTheme(Download))
            fill="#004c97"
            height="13px"
            width="10px"
          />
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={false}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    />
  </div>
</div>
`;

exports[`<NewSpeechModel /> Functionality: should call isValidated should set the model as valid 1`] = `
<div
  id="NewModel"
>
  <ul
    className="form-fields"
  >
    <li>
      <label>
        Model
      </label>
      <div
        style={
          Object {
            "color": "#313f54",
          }
        }
      >
         
      </div>
    </li>
    <li>
      <label>
        Version Description
      </label>
      <withTheme(Textarea)
        defaultValue=""
        disabled={false}
        name="modelDescription"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a description"
        styleOverride={
          Object {
            "width": "100%",
          }
        }
        type="text"
      />
    </li>
    <li>
      <MultipleCheckboxFilter
        allowEmpty={true}
        className="DatasetsFilter"
        disabled={false}
        label={
          <div
            style={
              Object {
                "color": "#727272",
              }
            }
          >
            Datasets*
          </div>
        }
        noOfColumns={3}
        onChange={[Function]}
        options={Array []}
        showAsGrid={false}
        showDropbox={false}
        type={null}
        value={Array []}
      />
    </li>
  </ul>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginBottom": "30px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "30px",
        "width": "400px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <Memo(withTheme(Download))
            fill="#004c97"
            height="13px"
            width="10px"
          />
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={false}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    />
  </div>
</div>
`;

exports[`<NewSpeechModel /> Functionality: should call onclickDownload should call getDataURL 1`] = `
<div
  id="NewModel"
>
  <ul
    className="form-fields"
  >
    <li>
      <label>
        Model
      </label>
      <div
        style={
          Object {
            "color": "#313f54",
          }
        }
      >
         
      </div>
    </li>
    <li>
      <label>
        Version Description
      </label>
      <withTheme(Textarea)
        defaultValue=""
        disabled={false}
        name="modelDescription"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a description"
        styleOverride={
          Object {
            "width": "100%",
          }
        }
        type="text"
      />
    </li>
    <li>
      <MultipleCheckboxFilter
        allowEmpty={true}
        className="DatasetsFilter"
        disabled={false}
        label={
          <div
            style={
              Object {
                "color": "#727272",
              }
            }
          >
            Datasets*
          </div>
        }
        noOfColumns={3}
        onChange={[Function]}
        options={Array []}
        showAsGrid={false}
        showDropbox={false}
        type={null}
        value={Array []}
      />
    </li>
  </ul>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginBottom": "30px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "30px",
        "width": "400px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <Memo(withTheme(Download))
            fill="#004c97"
            height="13px"
            width="10px"
          />
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={false}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    />
  </div>
</div>
`;

exports[`<NewSpeechModel /> Functionality: should call rejectFile should call onChange 1`] = `
<div
  id="NewModel"
>
  <ul
    className="form-fields"
  >
    <li>
      <label>
        Model
      </label>
      <div
        style={
          Object {
            "color": "#313f54",
          }
        }
      >
         
      </div>
    </li>
    <li>
      <label>
        Version Description
      </label>
      <withTheme(Textarea)
        defaultValue=""
        disabled={false}
        name="modelDescription"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a description"
        styleOverride={
          Object {
            "width": "100%",
          }
        }
        type="text"
      />
    </li>
    <li>
      <MultipleCheckboxFilter
        allowEmpty={true}
        className="DatasetsFilter"
        disabled={false}
        label={
          <div
            style={
              Object {
                "color": "#727272",
              }
            }
          >
            Datasets*
          </div>
        }
        noOfColumns={3}
        onChange={[Function]}
        options={Array []}
        showAsGrid={false}
        showDropbox={false}
        type={null}
        value={Array []}
      />
    </li>
  </ul>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginBottom": "30px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "30px",
        "width": "400px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <Memo(withTheme(Download))
            fill="#004c97"
            height="13px"
            width="10px"
          />
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={true}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    />
    <ValidationPlaceholder
      styleOverride={
        Object {
          "marginTop": "10px",
        }
      }
      validationMessage="File format or content is invalid. Please consult documentation."
    />
  </div>
</div>
`;

exports[`<NewSpeechModel /> Functionality: should call saveFile should upload the file successfully 1`] = `
<div
  id="NewModel"
>
  <ul
    className="form-fields"
  >
    <li>
      <label>
        Model
      </label>
      <div
        style={
          Object {
            "color": "#313f54",
          }
        }
      >
         
      </div>
    </li>
    <li>
      <label>
        Version Description
      </label>
      <withTheme(Textarea)
        defaultValue=""
        disabled={false}
        name="modelDescription"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a description"
        styleOverride={
          Object {
            "width": "100%",
          }
        }
        type="text"
      />
    </li>
    <li>
      <MultipleCheckboxFilter
        allowEmpty={true}
        className="DatasetsFilter"
        disabled={false}
        label={
          <div
            style={
              Object {
                "color": "#727272",
              }
            }
          >
            Datasets*
          </div>
        }
        noOfColumns={3}
        onChange={[Function]}
        options={Array []}
        showAsGrid={false}
        showDropbox={false}
        type={null}
        value={Array []}
      />
    </li>
  </ul>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginBottom": "30px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "30px",
        "width": "400px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <Memo(withTheme(Download))
            fill="#004c97"
            height="13px"
            width="10px"
          />
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={false}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    />
  </div>
</div>
`;

exports[`<NewSpeechModel /> Functionality: should call saveModelChanges should call saveModelChanges 1`] = `
<div
  id="NewModel"
>
  <ul
    className="form-fields"
  >
    <li>
      <label>
        Model
      </label>
      <div
        style={
          Object {
            "color": "#313f54",
          }
        }
      >
         
      </div>
    </li>
    <li>
      <label>
        Version Description
      </label>
      <withTheme(Textarea)
        defaultValue=""
        disabled={false}
        name="modelDescription"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a description"
        styleOverride={
          Object {
            "width": "100%",
          }
        }
        type="text"
      />
    </li>
    <li>
      <MultipleCheckboxFilter
        allowEmpty={true}
        className="DatasetsFilter"
        disabled={false}
        label={
          <div
            style={
              Object {
                "color": "#727272",
              }
            }
          >
            Datasets*
          </div>
        }
        noOfColumns={3}
        onChange={[Function]}
        options={Array []}
        showAsGrid={false}
        showDropbox={false}
        type={null}
        value={Array []}
      />
    </li>
  </ul>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginBottom": "30px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "30px",
        "width": "400px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <Memo(withTheme(Download))
            fill="#004c97"
            height="13px"
            width="10px"
          />
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={false}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    />
  </div>
</div>
`;

exports[`<NewSpeechModel /> Snapshots should match with basic props 1`] = `
<div
  id="NewModel"
>
  <ul
    className="form-fields"
  >
    <li>
      <label>
        Model
      </label>
      <div
        style={
          Object {
            "color": "#313f54",
          }
        }
      >
         
      </div>
    </li>
    <li>
      <label>
        Version Description
      </label>
      <withTheme(Textarea)
        defaultValue=""
        disabled={false}
        name="modelDescription"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Enter a description"
        styleOverride={
          Object {
            "width": "100%",
          }
        }
        type="text"
      />
    </li>
    <li>
      <MultipleCheckboxFilter
        allowEmpty={true}
        className="DatasetsFilter"
        disabled={false}
        label={
          <div
            style={
              Object {
                "color": "#727272",
              }
            }
          >
            Datasets*
          </div>
        }
        noOfColumns={3}
        onChange={[Function]}
        options={Array []}
        showAsGrid={false}
        showDropbox={false}
        type={null}
        value={Array []}
      />
    </li>
  </ul>
  <div
    id="speech-dropbox"
    style={
      Object {
        "marginBottom": "30px",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": "30px",
        "width": "400px",
      }
    }
  >
    <div
      style={
        Object {
          "paddingBottom": "20px",
        }
      }
    >
      <span
        style={
          Object {
            "color": "#6E6E6E",
            "fontSize": "14.5px",
            "fontWeight": "normal",
            "paddingBottom": "5px",
          }
        }
      >
        Upload word class file
      </span>
      <div
        style={
          Object {
            "display": "inline-flex",
            "float": "right",
            "icon": Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            },
            "label": Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            },
          }
        }
      >
        <span
          style={
            Object {
              "marginTop": "-3px",
              "paddingRight": "5px",
            }
          }
        >
          <Memo(withTheme(Download))
            fill="#004c97"
            height="13px"
            width="10px"
          />
        </span>
        <div
          onClick={[Function]}
          style={
            Object {
              "color": "#6E6E6E",
              "cursor": "pointer",
              "fontSize": "12.5px",
              "fontWeight": "normal",
              "textDecoration": "underline",
            }
          }
        >
           Download existing 
        </div>
      </div>
    </div>
    <Dropzone
      accept="text/plain"
      acceptStyle={
        Object {
          "borderColor": "#00e676",
        }
      }
      acceptedIcon={
        Object {
          "$$typeof": Symbol(react.memo),
          "compare": null,
          "type": [Function],
        }
      }
      activeStyle={
        Object {
          "borderColor": "#2196f3",
        }
      }
      baseStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fafafa",
          "borderColor": "#eeeeee",
          "borderRadius": 2,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "color": "#bdbdbd",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "height": "170px",
          "outline": "none",
          "padding": "20px",
          "transition": "border .24s ease-in-out",
        }
      }
      disabled={false}
      icon={[Function]}
      multiple={false}
      rejectFile={[Function]}
      rejectStyle={
        Object {
          "borderColor": "#ff1744",
        }
      }
      removeFile={false}
      saveFile={[Function]}
      shouldUpload={false}
      uploadHandler={[Function]}
    />
  </div>
</div>
`;
